--- TTA_Crypto_protocol.h.orig	2012-09-17 20:58:33.000000000 +0200
+++ TTA_Crypto_protocol.h	2015-07-21 09:52:55.846555194 +0200
@@ -73,6 +73,20 @@
 } TEE_CryptoData;
 
 /** Functions definition **/
+uint32_t retrieveUint32FromBuffer(char buffer[]);
+TEE_OperationHandle getOperation(uint32_t lopId); 
+TEE_Attribute* getAttribute(void);
+uint32_t getAttributeCount(void);
+void SetOperationToNull(uint32_t lopId);
+TEE_ObjectHandle getKey(uint32_t lkeyId);
+TEE_Result CmdDeriveKey(
+   void*       pSessionContext,
+   uint32_t    nParamTypes,
+   TEE_Param   pParams[4]);
+TEE_Result CmdGenerateRandom(
+   void*       pSessionContext,
+   uint32_t    nParamTypes,
+   TEE_Param   pParams[4]);
 TEE_Result CmdInitObjectWithKeys(
    void*       pSessionContext,
    uint32_t    nParamTypes,
@@ -109,6 +123,10 @@
    void*       pSessionContext,
    uint32_t    nParamTypes,
    TEE_Param   pParams[4]);
+TEE_Result CmdGetOperationInfoMultiple(
+   void*       pSessionContext,
+   uint32_t    nParamTypes,
+   TEE_Param   pParams[4]);
 TEE_Result CmdDigestUpdate(
    void*       pSessionContext,
    uint32_t    nParamTypes,
@@ -217,4 +235,5 @@
 #define CMD_LoadAttributes                   0x0001001C
 #define CMD_DeriveKey                        0x0001001D
 #define CMD_GenerateRandom                   0x0001001E
-                                                     
\ No newline at end of file
+#define CMD_GetOperationInfoMultiple         0x0001001F
+                                                     
