--- xtest_test.c	2015-06-12 09:22:13.169957949 +0200
+++ xtest_test.c.wgp	2015-06-11 15:39:51.883864032 +0200
@@ -21,6 +21,8 @@
 #include <ta_storage.h>
 #include <enc_fs_key_manager_test.h>
 #include <tee_api_defines.h>
+#include <tee_api_types.h>
+#include <TTA_DS_protocol.h>
 
 ADBG_ENUM_TABLE_DEFINE_BEGIN(TEEC_Result)
 ADBG_ENUM_TABLE_ENTRY(TEEC_SUCCESS),
@@ -74,4 +76,5 @@
 const TEEC_UUID sims_test_ta_uuid = TA_SIMS_TEST_UUID;
 const TEEC_UUID storage_ta_uuid = TA_STORAGE_UUID;
 const TEEC_UUID enc_fs_key_manager_test_ta_uuid = ENC_FS_KEY_MANAGER_TEST_UUID;
+const TEEC_UUID gp_tta_ds_uuid = TA_TTA_DS_UUID;
 
